### PR TITLE
CI: clang-format: update ubuntu, more generic names

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES
- CI: add pre-commit workflow with clang-format and whitespace checker

ENDRELEASENOTES

### CODE REVIEW

File: .github/workflows/clang-format-check.yml [hunk 1]
```diff
@@ -4,21 +4,21 @@ on: [push, pull_request]
...
+          -it --name k4_CI \
+          -v ${{ github.workspace }}:/Package \
           -v /cvmfs:/cvmfs:shared \
           -d ghcr.io/aidasoft/centos7:latest /bin/bash
 
     - name: Check Format
       run: |
-        docker exec k4E2L_CI /bin/bash -c 'source /cvmfs/sft.cern.ch/lcg/releases/clang/10.0.0-62e61/x86_64-centos7/setup.sh;\
-        clang-format -Werror --verbose --dry-run --style=file  $(find ./k4E2L/ -name "*.cpp" -o -name "*.h" );\
+        docker exec K4_CI /bin/bash -c 'source /cvmfs/sft.cern.ch/lcg/releases/clang/10.0.0-62e61/x86_64-centos7/setup.sh;\
```
Comment (andresailer) [inline .github/workflows/clang-format-check.yml:21]:
```suggestion
        docker exec k4_CI /bin/bash -c 'source /cvmfs/sft.cern.ch/lcg/releases/clang/10.0.0-62e61/x86_64-centos7/setup.sh;\
```

**Review by tmadlener — COMMENTED** [review]
Any reason to not run this via `pre-commit`?

c.f. https://github.com/key4hep/k4SimDelphes/blob/main/.github/workflows/pre-commit.yml

**Comment by andresailer** [comment]
Let me try with that!

**Review by tmadlener — COMMENTED** [review]
Still missing for pre-commit:
- config (e.g. https://github.com/key4hep/k4SimDelphes/blob/main/.pre-commit-config.yaml)
- clang-format-wrapper script (https://github.com/key4hep/k4SimDelphes/blob/main/.github/scripts/clang-format-hook)

**Review by tmadlener — APPROVED** [review]

**Pull request merged**